python_runner/runner.py

```diff
@@ -65,21 +65,12 @@ def callback(self, event_type, **data):
     def output(self, output_type, text, **extra):
         return self.output_buffer.put(output_type, text, **extra)
 
-    def snoop_config(self, out=None, color=False):
-        import snoop
-        if out is None:
-            from .snoop import SnoopStream
-            out = SnoopStream(self.output_buffer)
-        return snoop.Config(
-            columns=(),
-            out=out,
-            color=color,
-        )
-
-    def execute(self, code_obj, mode=None):  # noqa
+    def execute(self, code_obj, mode=None, snoop_config=None):  # noqa
         if mode == "snoop":
-            from .snoop import exec_snoop
-            exec_snoop(self, code_obj, config=self.snoop_config())
+            from .snoop import exec_snoop, SnoopStream
+            if snoop_config is None:
+               snoop_config = dict(out=SnoopStream(self.output_buffer), color=False)
```

Comment:
These individual values (plus `columns=()`) should be the defaults, and if snoop_config is specified it should be merged with this. The logic is still best in exec_snoop, but with these values, not the ones there.